src/Mvc/Mvc.NewtonsoftJson/test/NewtonsoftJsonInputFormatterTest.cs

```diff
@@ -211,6 +206,12 @@ public override Task JsonFormatter_EscapedKeys_Bracket()
         return base.JsonFormatter_EscapedKeys_Bracket();
     }
 
+    [Fact(Skip = "Expected: [0]['It\\'s a key'], Actual: [0]['It\\'s a key'].It's a key")]
```

Comment:
I discovered this issue while working on this. I'll resolve this or file another issue and reference it here before merging the PR.

`ErrorContext.Path` escapes the `'`, but `ErrorContext.Member` doesn't meaning the `!path.EndsWith()` checks don't cover this scenario. We could scan `ErrorContext.Member` for any `'` characters and manually escape it before doing the `!path.EndsWith()` checks, but that feels like playing whac-a-mole. I'm guessing there are even more edge cases the `!path.EndsWith()` checks don't cover.

I really want to get rid of the `addMember` logic altogether and have Json.NET either give us the path we want to begin with or expose an `ErrorContext.ErrorType` so we only do the `addMember` logic for missing required properties where we know the member is never in the path. 

This is what @JamesNK suggested on @dougbu's PR a while back, but it looks like it wasn't possible due to time constraints. See https://github.com/aspnet/Mvc/pull/8659#discussion_r229449516 Is this something we could do for .NET 7 @JamesNK?

Reply:
I remember that issue. It was rather hacky.

The improvement I mentioned needs the discussed feature added to Newtonsoft.Json. I haven't thought any more about adding it.

Reply:
> This is what @JamesNK suggested on @dougbu's PR a while back, but it looks like it wasn't possible due to time constraints. See https://github.com/aspnet/Mvc/pull/8659#discussion_r229449516 Is this something we could do for .NET 7 @JamesNK?

nit: The problem wasn't exactly time constraints; it was the lack of the feature @JamesNK mentioned (then and now).

> I remember that issue. It was rather hacky.

My bad. This fix is a bit more complicated and (unfortunately) not something I thought of last time around.